mne/chpi.py

```diff
@@ -469,14 +463,22 @@ def _setup_hpi_amplitude_fitting(info, t_window, remove_aliased=False,
         raise RuntimeError('Found HPI frequencies %s above the lowpass '
                            '(or Nyquist) frequency %0.1f'
                            % (hpi_freqs[~keepers].tolist(), highest))
-    if info['line_freq'] is not None:
-        line_freqs = np.arange(info['line_freq'], info['sfreq'] / 3.,
-                               info['line_freq'])
-    else:
-        line_freqs = np.zeros([0])
-    logger.info('Line interference frequencies: %s Hz'
-                % ' '.join(['%d' % lf for lf in line_freqs]))
-
+    # calculate optimal window length.
+    if isinstance(t_window, str):
+        if t_window != 'auto':
+            raise ValueError('t_window must be "auto" if a string, got %r'
+                             % (t_window,))
+        if len(hpi_freqs):
+            all_freqs = np.concatenate((hpi_freqs, line_freqs))
+            delta_freqs = np.diff(np.unique(all_freqs))
+            t_window = max(5. / all_freqs.min(), 1. / delta_freqs.min())
+        else:
+            t_window = 0.2
+    t_window = float(t_window)
+    if t_window <= 0:
+        raise ValueError('t_window (%s) must be > 0' % (t_window,))
+    logger.info('Using time window: %0.1f ms' % (1000 * t_window,))
```

Comment:
In a test that uses `test_move_anon_raw`, this logging line should have its value changed IIRC, because the line freq is 60 and the first cHPI freq is 83. So your updated test could be something like:
```
with catch_logging() as log:
    compute_chpi_...(..., t_window='auto', verbose=True)
assert '43.25 ms' in log.getvalue()
```
or something similar.